allensdk/internal/api/mesoscope_lims_api.py

```diff
@@ -0,0 +1,371 @@
+        for idx, trial in trial_df.iterrows():
+            rewards = trial["rewards"]  # as i write this there can only ever be one reward per trial
+            if rewards:
+                rewards_dict["volume"].append(rewards[0][0])
+                rewards_dict["timestamps"].append(rebase_function(rewards[0][1]))
+                rewards_dict["autorewarded"].append('auto_rewarded' in trial['trial_params'])
+        df = pd.DataFrame(rewards_dict).set_index('timestamps', drop=True)
+        return df
+
+    @memoize
+    def get_trials(self):
+        licks = self.get_licks()
+        behavior_stimulus_file = self.get_behavior_stimulus_file()
+        data = pd.read_pickle(behavior_stimulus_file)
+        rewards = self.get_rewards()
+        stimulus_timestamps_no_monitor_delay = self.get_sync_data()['stimulus_times_no_delay'][:-1]
+        rebase_function = get_stimulus_rebase_function(data, stimulus_timestamps_no_monitor_delay)
+        trial_df = get_trials(data, licks, rewards, rebase_function)
+        return trial_df
+
+    def get_sync_data(self):
+
+        # let's fix line labels where they are off #
+
+        sync_file = self.get_sync_file()
+        sync_dataset = SyncDataset(sync_file)
+        line_labels = sync_dataset.line_labels
+
+        wrong_labels = ['vsync_2p', 'stim_vsync', 'photodiode', 'acq_trigger', 'cam1', 'cam2',]
+        correct_labels = ['2p_vsync', '', 'stim_vsync', '', 'stim_photodiode', 'acq_trigger', '', '', 'cam1_exposure', 'cam2_exposure', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', '', 'lick_sensor']
+
+
+        if line_labels != correct_labels :
+            if all([line_labels[i] == '' for i in range(len(line_labels))]) == True :
+                logger.warning('Line labels are empty, replacing with defaults')
+                line_labels = correct_labels
+            else:
+                logger.warning('Some line labels are incorrect, renaming using defaults')
+                i = 0
+                while i < len(line_labels):
+                    line = line_labels[i]
+                    if line in wrong_labels :
+                        line_labels.pop(i)
+                        line_labels.insert(i, correct_labels[i])
+                    i +=1
+
+        sync_dataset.line_labels = line_labels
+        meta_data = sync_dataset.meta_data
+        sample_freq = meta_data['ni_daq']['counter_output_freq']
+
+        # 2P vsyncs
+        vs2p_r = sync_dataset.get_rising_edges('2p_vsync')
+        vs2p_f = sync_dataset.get_falling_edges('2p_vsync')  # new sync may be able to do units = 'sec', so conversion can be skipped
+        frames_2p = vs2p_r / sample_freq
+        vs2p_fsec = vs2p_f / sample_freq
+
+        # use rising edge for Scientifica and Mesoscope falling edge for Nikon http://confluence.corp.alleninstitute.org/display/IT/Ophys+Time+Sync
+        stimulus_times_no_monitor_delay = sync_dataset.get_rising_edges('stim_vsync') / sample_freq
```

Comment:
falling - also, I think Doug has fixed this in allensdk. Please see if the existing implementation does the trick and modify that.